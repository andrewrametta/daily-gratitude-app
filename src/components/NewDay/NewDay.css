h4 {
  color: var(--teal);
  text-align: center;
  margin-bottom: 10px;
}

h5 {
  font-size: 14px;
  text-align: center;
  color: rgb(84, 84, 84);
}

form .new-day-form {
  text-align: center;
}

.new-day {
  text-align: center;
  margin-bottom: 30px;
}

textarea {
  margin: 10px 10px;
  padding: 10px;
}

form textarea {
  border-radius: 10px;
}
