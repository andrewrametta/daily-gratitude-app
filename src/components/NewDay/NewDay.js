import React, { Component } from "react";
import AuthApiService from "../../services/auth-api-service";
import Context from "../../Context";

class NewDay extends Component {
  static contextType = Context;
  handleSubmit = (e) => {
    e.preventDefault();
    const { text1, text2, text3 } = e.target;
    AuthApiService.postDay({
      text1: text1.value,
      text2: text2.value,
      text3: text3.value,
    })
      .then((day) => {
        console.log(day);
        this.context.addNewDay(day);
        this.props.history.push("/dashboard");
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { days = [] } = this.context;
    return (
      <div>
        <header>
          {days.length === 0 ? <h4>Day 1</h4> : <h4>Day {days.length + 1}</h4>}
          <h5>Write 3 things you are grateful for in the text boxes below</h5>
        </header>
        <section>
          <form id="new-day" onSubmit={this.handleSubmit}>
            <section className="form-section overview-section">
              <label htmlFor="text1"></label>
              <textarea name="text1" rows="4" required></textarea>
            </section>
            <section className="form-section overview-section">
              <label htmlFor="text2"></label>
              <textarea name="text2" rows="4" required></textarea>
            </section>
            <section className="form-section overview-section">
              <label htmlFor="text3"></label>
              <textarea name="text3" rows="4" required></textarea>
              <button type="submit">Submit</button>
            </section>
          </form>
        </section>
      </div>
    );
  }
}

export default NewDay;
