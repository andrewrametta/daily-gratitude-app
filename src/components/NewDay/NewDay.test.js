import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import NewDay from "./NewDay";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe(`NewDay component`, () => {
  it("renders the complete page", () => {
    const wrapper = shallow(<NewDay />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
