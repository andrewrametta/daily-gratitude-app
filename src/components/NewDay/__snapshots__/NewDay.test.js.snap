// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewDay component renders the complete page 1`] = `
<Fragment>
  <h1>
    Day 1
  </h1>
  <h5>
    Write 3 things you are grateful for in the text boxes below
  </h5>
  <div
    className="new-day"
  >
    <form
      className="new-day-form"
      id="new-day"
      onSubmit={[Function]}
    >
      <section
        className="form-section overview-section"
      >
        <label
          htmlFor="text1"
        />
        <textarea
          name="text1"
          required={true}
          rows="4"
        />
      </section>
      <section
        className="form-section overview-section"
      >
        <label
          htmlFor="text2"
        />
        <textarea
          name="text2"
          required={true}
          rows="4"
        />
      </section>
      <section
        className="form-section overview-section"
      >
        <label
          htmlFor="text3"
        />
        <textarea
          name="text3"
          required={true}
          rows="4"
        />
        <button
          type="submit"
        >
          Submit
        </button>
      </section>
    </form>
  </div>
</Fragment>
`;
